int8 num
float32 target_speed
float32 current_speed
float32 ddist